WebHostLib/customserver.py

```diff
@@ -245,25 +245,19 @@ async def start_room(room_id):
             ctx.shutdown_task = asyncio.create_task(auto_shutdown(ctx, []))
             await ctx.shutdown_task
 
-            # ensure auto launch is on the same page in regard to room activity.
-            with db_session:
-                room: Room = Room.get(id=ctx.room_id)
-                room.last_activity = datetime.datetime.utcnow() - datetime.timedelta(seconds=room.timeout + 60)
-
         except (KeyboardInterrupt, SystemExit):
-            with db_session:
-                room = Room.get(id=room_id)
-                # ensure the Room does not spin up again on its own, minute of safety buffer
-                room.last_activity = datetime.datetime.utcnow() - datetime.timedelta(minutes=1, seconds=room.timeout)
+            pass
         except Exception:
             with db_session:
                 room = Room.get(id=room_id)
                 room.last_port = -1
-                # ensure the Room does not spin up again on its own, minute of safety buffer
-                room.last_activity = datetime.datetime.utcnow() - datetime.timedelta(minutes=1, seconds=room.timeout)
             raise
         finally:
+            with db_session:
+                # ensure the Room does not spin up again on its own, minute of safety buffer
+                room.last_activity = datetime.datetime.utcnow() - datetime.timedelta(minutes=1, seconds=room.timeout)
```

Comment:
Is it safe to reuse room in a new db_session?  In the old code we selected room from id again.

Reply:
probably not, fixed to prevent future disappointment.

Reply:
FTR, it looks like it would be safe assuming the room did not get deleted in the meantime. But this may very well be an implementation detail we don't want to rely on.